src/luascript.cpp

```diff
@@ -13777,66 +13812,952 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
+	Spell* spell = dynamic_cast<InstantSpell*>(getUserdata<InstantSpell>(L, 1));
+	if (spell) {
+		if (spell->spellType == SPELL_UNDEFINED) {
+			spell = dynamic_cast<RuneSpell*>(getUserdata<RuneSpell>(L, 1));
+		}
+
+		if (lua_gettop(L) == 1) {
+			pushBoolean(L, spell->getBlockingSolid());
+			pushBoolean(L, spell->getBlockingCreature());
+			return 2;
+		} else {
+			spell->setBlockingSolid(getBoolean(L, 2));
+			spell->setBlockingCreature(getBoolean(L, 3));
+			pushBoolean(L, true);
+		}
+	} else {
+		lua_pushnil(L);
+	}
+	return 1;
+}
+
+int LuaScriptInterface::luaSpellAggressive(lua_State* L)
+{
+	// spell:isAggressive(bool)
+	Spell* spell = dynamic_cast<InstantSpell*>(getUserdata<InstantSpell>(L, 1));
+	if (spell) {
+		if (spell->spellType == SPELL_UNDEFINED) {
+			spell = dynamic_cast<RuneSpell*>(getUserdata<RuneSpell>(L, 1));
+		}
+
+		if (lua_gettop(L) == 1) {
+			pushBoolean(L, spell->getAggressive());
+		} else {
+			spell->setAggressive(getBoolean(L, 2));
+			pushBoolean(L, true);
+		}
+	} else {
+		lua_pushnil(L);
+	}
+	return 1;
+}
+
+int LuaScriptInterface::luaSpellVocation(lua_State* L)
+{
+	// spell:vocation(vocation[, showInDescription = false)
+	Spell* spell = dynamic_cast<InstantSpell*>(getUserdata<InstantSpell>(L, 1));
+	if (spell) {
+		if (spell->spellType == SPELL_UNDEFINED) {
+			spell = dynamic_cast<RuneSpell*>(getUserdata<RuneSpell>(L, 1));
+		}
+
+		if (lua_gettop(L) == 1) {
+			lua_createtable(L, 0, 0);
+			auto it = 0;
+			for (auto voc : spell->getVocMap()) {
+				++it;
+				std::string s = std::to_string(it);
+				char const *pchar = s.c_str();
+				std::string name = g_vocations.getVocation(voc.first)->getVocName();
+				setField(L, pchar, name);
+			}
+			setMetatable(L, -1, "Spell");
+		} else {
+			int32_t vocationId = g_vocations.getVocationId(getString(L, 2));
+			if (vocationId != -1) {
+				bool showInDescription = false;
+				if (lua_gettop(L) == 3) {
+					showInDescription = getBoolean(L, 3);
+				}
+				spell->addVocMap(vocationId, showInDescription);
+				pushBoolean(L, true);
+			}
+		}
+	} else {
+		lua_pushnil(L);
+	}
+	return 1;
+}
+
+// only for InstantSpells
+int LuaScriptInterface::luaSpellWords(lua_State* L)
+{
+	// spell:words(words[, separator = ""])
+	InstantSpell* spell = getUserdata<InstantSpell>(L, 1);
+	if (spell) {
+		// if spell == SPELL_UNDEFINED, it means that this actually is no InstantSpell, so we return nil
+		if (spell->spellType == SPELL_UNDEFINED) {
+			lua_pushnil(L);
+			return 1;
+		}
+
+		if (lua_gettop(L) == 1) {
+			pushString(L, spell->getWords());
+			pushString(L, spell->getSeparator());
+			return 2;
+		} else {
+			std::string sep = "";
+			if (lua_gettop(L) == 3) {
+				sep = getString(L, 3);
+			}
+			spell->setWords(getString(L, 2));
+			spell->setSeparator(sep);
+			pushBoolean(L, true);
+		}
+	} else {
+		lua_pushnil(L);
+	}
+	return 1;
+}
+
+// only for InstantSpells
+int LuaScriptInterface::luaSpellNeedDirection(lua_State* L)
+{
```

Comment:
~This should also probably be renamed to spell:castOnDirection because it is not that it needs direction but instead asserting (imperative) that it will cast the spell towards the direction the caster is facing, so it is not like it is a requirement.~

Since this works both ways ``getting`` and ``setting`` it only makes sense to rename this and all others functions to just their main words, like ``spell:direction()``, ``spell:params()``, ``spell:blockingWalls()`` etc, like you mentioned in an issue if I recall correctly.

It would not be pretty to rename it to something like ``spell:setDirectionOrGetDirection`` haha.

Reply:
The problem would be that it gives the implication that something like `:direction` would set or get the `direction` instead of expecting/returning a `boolean` value.
`is` or `need` tells pretty good what we expect to get/set from the function.
We should rather rename all boolean functions to this standard and have it as get/set already.